llvm/lib/Target/AArch64/AsmParser/AArch64AsmParser.cpp

```diff
@@ -7234,6 +7236,8 @@ bool AArch64AsmParser::parseDirectiveArchExtension(SMLoc L) {
     STI.ClearFeatureBitsTransitively(It->Features);
   FeatureBitset Features = ComputeAvailableFeatures(STI.getFeatureBits());
   setAvailableFeatures(Features);
+
+  getTargetStreamer().emitDirectiveArchExtension(Name);
```

Comment:
This is wrong, as `Name` might have been updated on (new) line 7222. This would be clear with a `.arch_extension nolse` test.

Reply:
Thank you for catching that. I have updated the code so that it emits the full name from before the `no` prefix is stripped, along with a test. Let me know if you'd like further changes.